crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -188,6 +209,66 @@ where
         Ok(ForkchoiceUpdated::new(status))
     }
 
+    /// Validates the payload attributes with respect to the header and fork choice state.
+    fn process_payload_attributes(
+        &self,
+        attrs: PayloadAttributes,
+        header: Header,
+        state: ForkchoiceState,
+    ) -> Result<ForkchoiceUpdated, BeaconEngineError> {
+        // 7. Client software MUST ensure that payloadAttributes.timestamp is
+        //    greater than timestamp of a block referenced by
+        //    forkchoiceState.headBlockHash. If this condition isn't held client
+        //    software MUST respond with -38003: `Invalid payload attributes` and
+        //    MUST NOT begin a payload build process. In such an event, the
+        //    forkchoiceState update MUST NOT be rolled back.
+        if attrs.timestamp <= header.timestamp.into() {
+            return Ok(ForkchoiceUpdated::new(PayloadStatus::from_status(
+                PayloadStatusEnum::Invalid {
+                    validation_error: BeaconEngineError::InvalidPayloadAttributes.to_string(),
+                },
+            )))
+        }
+
+        // 8. Client software MUST begin a payload build process building on top of
+        //    forkchoiceState.headBlockHash and identified via buildProcessId value
+        //    if payloadAttributes is not null and the forkchoice state has been
+        //    updated successfully. The build process is specified in the Payload
+        //    building section.
+        let payload_id = self.payload_store.new_payload(header.parent_hash, attrs)?;
+
+        // Client software MUST respond to this method call in the following way:
+        // {
+        //      payloadStatus: {
+        //          status: VALID,
+        //          latestValidHash: forkchoiceState.headBlockHash,
+        //          validationError: null
+        //      },
+        //      payloadId: buildProcessId
+        // }
+        //
+        // if the payload is deemed VALID and the build process has begun.
+        Ok(ForkchoiceUpdated::new(PayloadStatus::new(
+            PayloadStatusEnum::Valid,
+            Some(state.head_block_hash),
+        ))
+        .with_payload_id(payload_id))
+    }
+
+    /// Called to receive the execution payload associated with a payload build process.
+    pub fn on_get_payload(&self, payload_id: H64) -> Result<ExecutionPayload, BeaconEngineError> {
+        // TODO: Client software SHOULD stop the updating process when either a call to
+        // engine_getPayload with the build process's payloadId is made or SECONDS_PER_SLOT (12s in
+        // the Mainnet configuration) have passed since the point in time identified by the
+        // timestamp parameter.
+
+        // for now just return the output from the payload store
```

Comment:
these are useful comments.

One thing I'm really happy with is the delegation that's happening 
RPC -> beacon engine -> payload handler